### DISCUSSION TITLE
z.string().datetime() invalid month validation

### DISCUSSION THREAD

**Discussion by FranciscoGileno** [colinhacks/zod#1931] [Q&A] [answered]
```ts
const schema = z.string().datetime({ offset: true })
console.log(schema.safeParse('2022-15-14T12:00:00-03:00')) //OBS: The month is invalid.
// {
//     success: true,
//     data: '2022-15-14T12:00:00-03:00'
// }
```

Is this a bug?

https://codesandbox.io/s/festive-moore-vc1mkd?file=/src/index.js

**Comment by JacobWeisenburger** [comment, marked as answer]
`z.string().datetime()` doesn't validate the date, it just checks that it's a valid iso datetime string using regex.

```ts
const schema = z.string().datetime({ offset: true })
console.log(schema.safeParse('0000-00-00T00:00:00-00:00'))
// {
//     success: true,
//     data: '0000-00-00T00:00:00-00:00'
// }
```

To validate the date, do this:

```ts
const schema = z.string().datetime({ offset: true }).pipe(z.coerce.date())

console.log(schema.safeParse('2023-01-25T00:00:00-00:00'))
// { success: true, data: 2023-01-25T00:00:00.000Z }

const result = schema.safeParse('0000-00-00T00:00:00-00:00')
!result.success && console.log(result.error.issues)
// [ { code: 'invalid_date', path: [], message: 'Invalid date' } ]
```

**Reply by FranciscoGileno** [reply]
Thanks Jacob! Great answer.

**Reply by FranciscoGileno** [reply]
There is only one issue with the @JacobWeisenburger answer.  The result changes the timezone. 

Not sure how to keep the original value yet. 

```ts
const schema = z.string().datetime({ offset: true }).pipe(z.coerce.date())

const result = schema.safeParse("2022-12-14T12:00:00-03:00")

// { "success": true, "data": "2022-12-14T15:00:00.000Z" }
```

**Reply by JacobWeisenburger** [reply]
Unfortunately this has nothing to do with Zod. This is just how the Date constructor works in JavaScript.

```ts
console.log(new Date('2022-12-14T12:00:00-03:00'))
// 2022-12-14T15:00:00.000Z
```

**Reply by BrandonNoad** [reply]
> it just checks that it's a valid iso datetime string using regex

I don't think a valid ISO 8601 datetime string allows invalid years, months, days, etc. So there seems to be a bug in the regex.

If the current behaviour is the intended behaviour, then I think the documentation should mention that.

**Reply by JacobWeisenburger** [reply]
PRs are always welcome.
